config/hwloc_internal.m4

```diff
@@ -355,6 +355,32 @@ EOF
       AC_CHECK_FUNCS([clock_gettime])
     ])
 
+    # Check for ptrace support
+    AC_CHECK_HEADERS([sys/ptrace.h], [
+      AC_CHECK_FUNCS([ptrace],[hwloc_have_ptrace=yes], [hwloc_have_ptrace=no])
+      AC_COMPILE_IFELSE([AC_LANG_SOURCE([[ 
+        #include "sys/ptrace.h"
+        int main(void){ return ptrace(PTRACE_SEIZE, -1, 0, (void*)(PTRACE_O_TRACECLONE|PTRACE_O_TRACEFORK)); }
+      ]])],, [hwloc_have_ptrace=no])
+    ])
+    AM_CONDITIONAL([HWLOC_HAVE_PTRACE],[ test "x$hwloc_have_ptrace" = xyes ])
+    if test x$hwloc_have_ptrace = xyes; then
+       AC_DEFINE([HWLOC_HAVE_PTRACE], [1], [ptrace is present and supportes PTRACE_SEIZE])
```

Comment:
It is safer to *always* `AC_DEFINE[_UNQUOTED]` a value -- i.e., always define it to 0 or 1 (vs. undef or 1).  

Specifically, if you go the "undef or 1" route, your C code looks like this:

```c
#ifdef HWLOC_HAVF_PTRACE
// do things with ptrace
```

But if you go the "always DEFINE to 0 or 1" route, your C code looks like this:

```c
#if HWLOC_HAVF_PTRACE
// do things with ptrace
```

I.e., `#if` vs. `#ifdef`.

The reason `#if` is better is because the compiler/preprocessor will tell you if you have a typo in your name, whereas `#ifdef` will not.

Did you notice that I misspelled `HWLOC_HAVE_PTRACE`, above?  The first code -- with `#ifdef` -- will silently always be false.  The second code -- with `#if` -- will give a compiler warning about an undefined preprocessor macro (`HWLOC_HAVF_PTRACE`), thereby alerting you to your typo.  Defensive coding === good.

Meaning: please always define logical preprocessor macros to 0 or 1 -- not undef-vs-1.

Thanks.

Reply:
Thanks for the tip. I will change that.

Reply:
Done see #https://github.com/open-mpi/hwloc/pull/359/commits/a5a097d6cc8f762fbd2d32a88d9f977c61f822e9